### PR TITLE
Align Radio to spec and add Sage Rails component

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description
- [x] - Align radio to spec
- [x] - add Rails Radio Component

### Screenshots

|  before  |  after  |
|--------|--------|
|Screen Shot 2020-10-19 at 6 13 41 PM|Screen Shot 2020-10-19 at 6 11 18 PM|

### Steps for testing
1. Visit the Sage Rails Radio page:  http://localhost:4000/pages/element/radio

### CODE REVIEW

**Review by pixelflips — APPROVED** [review]
Looks like a few of the files are missing a new line at the end. Which is probably causing the linting errors. Other than that, nice work! LGTM! 👍

File: lib/sage_rails/app/views/sage_components/_sage_radio.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div class="sage-radio<%= component.has_error ? " sage-radio--error" : "" %>">
+  <input 
+    id="<%= component.id %>" 
+    type="radio" 
+    class="sage-radio__input" 
+    name="<%= component.name %>" 
+    value="<%= component.value %>"
+    <%= component.checked ? " checked" : "" %>
+    <%= component.disabled ? " disabled" : "" %>
+    <%= component.required ? " required" : "" %>
```
Comment (joe-dakroub) [inline lib/sage_rails/app/views/sage_components/_sage_radio.html.erb]: Nice!

**Review by joe-dakroub — APPROVED** [review]
Everything is looking good; a much-needed enhancement. Nice work! 👍

File: app/views/examples/elements/radio/_preview.html.erb [hunk 2]
```diff
@@ -1,54 +1,58 @@
...
-id: "r2",
-name: "radio1",
-value: "2",
-label_text: "Option 2",
-checked: false,
-disabled: false,
-required: false,
-has_error: false
-%>
+<%= sage_component SageRadio, {
```
Comment (philschanely) [inline app/views/examples/elements/radio/_preview.html.erb:2]: In the design of the rails components I'd love to see any of these that are optional or have default values be able to be omitted from the preview documentation to slim this down. 🙏

File: lib/sage_rails/app/views/sage_components/_sage_radio.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<div class="sage-radio<%= component.has_error ? " sage-radio--error" : "" %>">
+  <input 
+    id="<%= component.id %>" 
+    type="radio" 
+    class="sage-radio__input" 
+    name="<%= component.name %>" 
+    value="<%= component.value %>"
+    <%= component.checked ? " checked" : "" %>
```
Comment (philschanely) [inline lib/sage_rails/app/views/sage_components/_sage_radio.html.erb]: FYI you can also do:
```
<%= "checked" if component.checked %>
```
if you don't need the ternaray for a default value. Also the initial space may be able to be omitted when each is on its own line like this?

**Review by philschanely — APPROVED** [review]
In great shape! Just a few suggestions to wrap up below.

File: app/views/examples/elements/radio/_preview.html.erb (hunk 2 above)
Comment (QuintonJason) [reply app/views/examples/elements/radio/_preview.html.erb:2]: Will look into this

**Pull request merged**